src/System.Drawing.Common/tests/IconTests.cs

```diff
@@ -246,7 +246,8 @@ public void Ctor_NullIcon_ThrowsArgumentNullException()
             AssertExtensions.Throws<ArgumentNullException, ArgumentException>("original", null, () => new Icon((Icon)null, new Size(32, 32)));
         }
 
-        [ActiveIssue(20884, TestPlatforms.AnyUnix)]
+        // libgdiplus causes a segfault when given an invalid Icon handle.
```

Comment:
Is this by-design? Is there a way to avoid this?

Reply:
I can't think of any to accomplish it without touching the native code. The reality is that we're dealing with native code here, and by calling `Icon.FromHandle`, we are essentially telling the native library to treat some arbitrary value as an Icon handle. Unless the native code had tracked every handle it issued and validated that you gave it a real, currently-valid handle, then segfaults aren't really possible to avoid. We could hard-code 0 and 1 as invalid handles, but that doesn't really help us all that much in the grand scheme of things.